runtime/list.go

```diff
@@ -170,6 +170,40 @@ func listCount(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return seqCount(f, args[0], args[1])
 }
 
+func listRemove(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if raised := checkMethodArgs(f, "remove", args, ListType, ObjectType); raised != nil {
+		return nil, raised
+	}
+	l := toListUnsafe(args[0])
+	l.mutex.Lock()
+	value := args[1]
+	found := false
+	var raised *BaseException
+	for i, elem := range l.elems {
+		var eq *Object
+		eq, raised = Eq(f, elem, value)
+		if raised != nil {
+			break
+		}
+		found, raised = IsTrue(f, eq)
+		if raised != nil {
+			break
+		}
+		if found {
+			l.elems = append(l.elems[:i], l.elems[i+1:]...)
+			break
+		}
+	}
+	l.mutex.Unlock()
+	if raised != nil {
+		return nil, raised
+	}
+	if found {
```

Comment:
I would do the `if !found {` condition here so that the default return at the bottom of the function is "success".